apps/server/src/modules/h5p-editor/temporary-file-storage/temporary-file-storage.ts

```diff
@@ -105,39 +57,23 @@ export class TemporaryFileStorage implements ITemporaryFileStorage {
 		rangeEnd?: number | undefined
 	): Promise<Readable> {
 		this.checkFilename(filename);
-		this.checkFilename(user.id);
-		const file = await open(join(this.path, user.id, filename)); // file is auto closed
-		const stream = file.createReadStream({
-			start: rangeStart,
-			end: rangeEnd,
-		});
-		return stream;
+		const tempFile = await this.repo.findByPath(user.id, filename);
+		const path = join(user.id, filename);
+		if (rangeStart === undefined) {
+			if (rangeEnd === undefined) {
+				return (await this.s3Client.get(path)).data;
+			}
+			rangeStart = 0;
+		} else if (rangeEnd === undefined) {
+			rangeEnd = tempFile.size - 1;
+		}
+		return (await this.s3Client.get(path, `${rangeStart}-${rangeEnd}`)).data;
 	}
 
 	public async listFiles(user?: IUser | undefined): Promise<ITemporaryFile[]> {
-		if (user) {
-			this.checkFilename(user.id);
-			await access(join(this.path, user.id));
-		}
-		const users: Array<string> = user ? [user.id] : await readdir(this.path);
-		const tempFiles: Array<ITemporaryFile> = [];
-		for (const currentUser of users) {
-			// no-await-in-loop es-lint check disabled here
-			// user by user should be okay here, otherwise there could be thousands of promises at once
-			// list ALL files is only needed for clean up job
-			const userpath = join(this.path, currentUser);
-			const userFiles: string[] = [];
-			for await (const file of this.readdirFilesRecursive(join(this.path, currentUser))) {
-				userFiles.push(relative(userpath, file));
-			}
-			const newTempFiles = await Promise.all(
-				userFiles
-					.filter((userFile) => !userFile.endsWith('.meta'))
-					.map(async (userFile) => this.getFileInfo(userFile, currentUser))
-			);
-			tempFiles.push(...newTempFiles);
-		}
-		return tempFiles;
+		// method is expected to support listing all files in database
+		// this is only needed for cleanup, so it is optimized to only return expired ones
+		return user ? this.repo.findByUser(user.id) : this.repo.findExpired();
```

Comment:
I think these should be two methods "listFilesForUser" and "listExpiredFiles". But I don't really get the comment. What do you mean by "expected"? By whom? And is the user part needed or not? If not remove it.